### ISSUE TITLE
Should we reduce the number of `timber/twig/…` filters?

### ISSUE THREAD

**Issue by gchtr** [timber/timber#2284]
After the issue in #2276, I was thinking about whether there’s a specific reason why we have so many different `timber/twig/…` filters that all do practically the same?

- `timber/twig/filters`
- `timber/twig/functions`
- `timber/twig/escapers`
- `timber/loader/twig`
- `timber/twig`

Wouldn’t it be enough if we only had `timber/twig`? The only reason I can think of now is to control the order in which functionality is added. When we as developers add our own functionality, we only use `timber/twig`, even if we add functions or filters.

Maybe only filters would be enough? Something like:

- `timber/twig/core` or `timber/twig/internal`
- `timber/twig`

Or we could also use the priority parameter in `add_filter()`:

```php
add_filter( 'timber/twig', function( $twig ) {
	// Internal stuff here.

    return $twig;
}, 9 );
```

@jarednova Do you have any insight on the history for this? For example, the `timber/twig/functions` filter was added in https://github.com/timber/timber/pull/1391/commits/21464d37dfc996338862461e7911d4841f171944, but I don’t know if it was just to continue the existing pattern or if there’s something I’m missing.

**Comment by jarednova** [comment]
I can't claim any grand plan for the exact organization. My recollection is that this set:

* `timber/twig/filters`
* `timber/twig/functions`
* `timber/twig/escapers`

... was added when doing a big re-organization of the `Timber\Twig` class with the thought of "hey, whatabout all these filters?" Well, it definitely succeeded in creating filters!

I'd agree that we could probably lose or drop these in a subsequent version and stick with something like `timber/twig` as what we emphasize to use. Of course there'd be a fair amount of effort to untangle and properly deprecate

**Comment by jarednova** [comment]
... and being very selective when it comes to adding any _new_ filters. There's an attractive elegance to sticking them in a bunch of places. But as #2276 highlights, those additions come with unforeseen risks and potential for mayhem.

**Comment by gchtr** [comment]
> I'd agree that we could probably lose or drop these in a subsequent version and stick with something like `timber/twig` as what we emphasize to use. Of course there'd be a fair amount of effort to untangle and properly deprecate

I agree. I moved this to be resolved in a future version.

**Comment by gchtr** [comment]
Resolved in #2419.
